client/client.go

```diff
@@ -251,3 +266,27 @@ func getDefaultAuthOpts(username, password string, enableTLS bool) []grpc.DialOp
 	)
 	return defaultOpts
 }
+
+func parseURI(uri string) (string, bool, error) {
+	hasPrefix := false
+	inSecure := false
+	if strings.HasPrefix(uri, "https://") {
+		inSecure = true
+		hasPrefix = true
+	}
+
+	if strings.HasPrefix(uri, "http://") {
```

Comment:
If feels weird that using `http://` as schema for uri while we are actually using grpc